api4/channel.go

```diff
@@ -1753,3 +1754,70 @@ func patchChannelModerations(c *Context, w http.ResponseWriter, r *http.Request)
 	auditRec.Success()
 	w.Write(b)
 }
+
+func moveChannel(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireChannelId()
+	if c.Err != nil {
+		return
+	}
+
+	channel, err := c.App.GetChannel(c.Params.ChannelId)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	props := model.StringInterfaceFromJson(r.Body)
+	teamId, ok := props["team_id"].(string)
+	if !ok {
+		c.SetInvalidParam("team_id")
+		return
+	}
+
+	team, err := c.App.GetTeam(teamId)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	var removeDeactivatedMembers bool
+	val, ok := props["removeDeactivatedMembers"].(string)
+	if ok {
+		removeDeactivatedMembers = val == "true"
```

Comment:
Does this need to be optional, or should we just always do this?

Reply:
Though I don't have strong opinions on this, I wonder why this is a part of moving a channel in the first place.

Reply:
If there are users in the source team that are not in the target team then you can't move the channel. If there are deactivated users in the source team that aren't in the target team, you can't remove them from the channel through the UI and you can't it makes no sense to add them to the target team. If it wasn't for this functionality, it would be impossible to move channels in those cases.

Reply:
I see, makes sense. We can remove deactivated users by default. I'm ok with that if this logic is in the `Api` layer. IMO we should divide this logic into two different methods in the `App` layer(`RemoveDeactivatedUsers `and `MoveChannel`), and use these together in the `Api` layer.

Reply:
Sounds good.